src/cc_catalog_airflow/dags/provider_api_scripts/phylopic.py

```diff
@@ -0,0 +1,357 @@
+    foreign_id = img_URL
+    foreign_url = '{}/image/{}'.format(base_URL, _uuid)
+
+    first_name = result.get('submitter', {}).get('firstName')
+    last_name = result.get('submitter', {}).get('lastName')
+    creator = '{} {}'.format(first_name, last_name).strip()
+
+    taxa = result.get('taxa', [])
+    # [0].get('canonicalName', {}).get('string')
+    taxa_list = None
+    if taxa:
+        taxa = list(filter(
+            lambda x: x.get('canonicalName') is not None, taxa))
+        taxa_list = list(
+            map(lambda x: x.get('canonicalName', {}).get('string', ''), taxa))
+
+    if taxa_list:
+        title = taxa_list[0]
+
+        if len(taxa_list) > 1:
+            meta_data['taxa'] = taxa_list
+
+    if result.get('credit'):
+        meta_data['credit_line'] = result.get('credit').strip()
+        meta_data['pub_date'] = result.get('submitted').strip()
+
+    return [
+            foreign_id, foreign_url, img_URL,
+            thumbnail if thumbnail else '\\N',
+            str(width) if width else '\\N',
+            str(height) if height else '\\N',
+            license_url,
+            creator if creator else '\\N',
+            title if title else '\\N',
+            json.dumps(
+                meta_data, ensure_ascii=False) if bool(meta_data) else '\\N',
+            '\\N', 'f', 'phylopic', 'phylopic'
+        ]
+
+
+def _get_image_info(result):
+    base_URL = 'http://phylopic.org'
+    img_URL = ''
+    thumbnail = ''
+    width = ''
+    height = ''
+
+    image_info = result.get('pngFiles')
+    if image_info:
+        img = list(filter(lambda x: (
+            int(str(x.get('width', '0'))) >= 257), image_info))
+        img = sorted(img, key=lambda x: x['width'], reverse=True)
+        thb = list(filter(lambda x: str(
+            x.get('width', '')) == '256', image_info))
+
+        if img:
+            img_URL = img[0].get('url')
+            img_URL = '{}{}'.format(base_URL, img_URL)
+            width = img[0].get('width')
+            height = img[0].get('height')
+
+            if thb:
+                thumbnail_info = thb[0].get('url')
+                if thumbnail_info:
+                    thumbnail = '{}{}'.format(base_URL, thumbnail_info)
+
+    if not img_URL:
+        return None
+    else:
+        return [img_URL, width, height, thumbnail]
+
+
+def _get_response_json(
+        endpoint=ENDPOINT,
+        retries=0,
+):
+    response_json = None
+
+    if retries < 0:
+        logger.error('No retries remaining.  Failure.')
+        raise Exception('Retries exceeded')
+
+    response = delayed_requester.get(
+        endpoint,
+        timeout=60
+    )
+    if response is not None and response.status_code == 200:
+        try:
+            response_json = response.json()
+        except Exception as e:
+            logger.warning(f'Could not get response_json.\n{e}')
+            response_json = None
+
+    if (
+            response_json is None
+            or response_json.get('error') is not None
```

Comment:
Are you completely sure that 'error' can appear in the response from PhyloPic?  I think this code is specific to another provider.